common/models/user.js

```diff
@@ -247,6 +247,12 @@ module.exports = function(User) {
             debug('An error is reported from User.hasPassword: %j', err);
             fn(defaultError);
           } else if (isMatch) {
+            if ((credentials.password).length > MAX_PASSWORD_LENGTH) {
```

Comment:
I think this validation should be called much earlier, just before calling `self.findOne`.